experiments/kompanion/cmd/export/export.go

```diff
@@ -90,10 +95,110 @@ func init() {
 
 	ExportCmd.Flags().StringArrayVarP(&targetObjects, targetObjectsFlag, "", []string{}, "object name prefix to target. Targets all if empty. Can be specified multiple times.")
 	ExportCmd.Flags().StringArrayVarP(&ignoreObjects, ignoreObjectsFlag, "", []string{}, "object name prefix to ignore. Excludes nothing if empty. Can be specified multiple times.")
+
+	ExportCmd.Flags().IntVarP(&workerRountines, workerRoutinesFlag, "", 10, "Configure the number of worker rountines to export namespaces with. Defaults to 10. ")
+}
+
+var (
+	reportName string
+	resources  []*metav1.APIResource
+)
+
+// tracks the namespaces to be exported.
+// thread safe.
+type namespacesWorkLog struct {
+	mu         sync.Mutex
+	namespaces []string // will be treated as a FIFO queue
+}
+
+var workLog *namespacesWorkLog
+
+func (n *namespacesWorkLog) Done() bool {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	return len(n.namespaces) == 0
+}
+
+func (n *namespacesWorkLog) GetWork() string {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	if len(n.namespaces) == 0 {
+		return ""
+	}
+
+	workItem, remainig := n.namespaces[0], n.namespaces[1:]
+	n.namespaces = remainig
+
+	return workItem
+}
+
+func processNamespace(namespaceName string, dynamicClient *dynamic.DynamicClient) {
+	log.Printf("Processing namespace: %s", namespaceName)
+	if shouldExclude(namespaceName, ignoreNamespaces, targetNamespaces) {
+		return
+	}
+
+	if err := os.Mkdir(filepath.Join(reportName, namespaceName), os.ModePerm); err != nil {
```

Comment:
Nit: I would use 0700 or 0755 (depending on whether it was sensitive).  I thought ModePerm was a mask.  Are you hoping that the umask makes this reasonable?

(You might also consider streaming directly into the tar.gz or zip file)

Reply:
Hm I thought `ModePerm` was o0777  as per https://pkg.go.dev/os but maybe i'm reading that wrong. Let's go w 0700 for now. 

Let me address the streaming in a separate PR since I'm also working on the cleanup and that will make cleaning up moot 😛 .